types/axe-webdriverjs/index.d.ts

```diff
@@ -74,7 +74,9 @@ export interface AxeBuilder {
      * Perform analysis and retrieve results.
      * @param callback   Function to execute when analysis completes.
      */
-    analyze(callback: (results: AxeAnalysis) => void): Promise<AxeAnalysis>;
+    analyze(
+        callback?: (err: Error, results: AxeAnalysis) => void
```

Comment:
The `err` param may be `null` here. I think this should be:

```ts
analyze(
  callback?: (err?: Error, result: AxeAnalysis) => void)
) : Promise<AxeAnalysis>
```

Also worth noting that we did not make a breaking change here, so the "old style" callback is still supported.

Reply:
@stephenmathieson Thanks for taking a look at my PR! I made an update to make the function signature this:

```javascript
analyze(
  callback?: (err: Error | null, result: AxeAnalysis) => void)
) : Promise<AxeAnalysis>
```

This addresses the fact that the error can be null. This does not address error still technically being an optional argument even though it is deprecated. The issue I ran into is TypeScript does not allow optional arguments before required ones. As far as I can tell this isn't something that you can tell it to ignore. 

Since using the old API results in a deprecation warning, do you think it's fine to mark it as required?

Reply:
I still think this is an improvement, because the current types force you to use the old API which can crash the process.

If anyone else has a workaround for that issue I'd be happy to try it though.